apollo-federation/src/merger/merger.rs

```diff
@@ -1028,8 +1029,124 @@ impl Merger {
         Ok(!source_as_entity.is_empty())
     }
 
-    pub(crate) fn merge_interface(&mut self, _itf: InterfaceTypeDefinitionPosition) {
-        todo!("Implement merge_interface")
+    fn validate_interface_keys(
+        &mut self,
+        sources: &Sources<Subgraph<Validated>>,
+        dest: &InterfaceTypeDefinitionPosition,
+    ) -> Result<bool, FederationError> {
+        let supergraph_implementations = self.merged.possible_runtime_types(dest.clone().into())?;
+
+        let mut has_key = false;
+        for (idx, source) in sources.iter() {
+            if source.is_none()
```

Comment:
instead of doing the unwrapping below you could split this check into two and extract subgraph in first check

```rs
let Some(subgraph) = source else {
  continue;
};
```